File: Frontend/src/app/room.ts
export class Room {
    id: number;
    roomNumber: number;
    roomType: string;
    price: number;
    available: boolean;

    constructor() {
        this.id = 0; // Example default value
        this.roomNumber = 0; // Example default value
        this.roomType = "Single"; // Example default value
        this.price = 0; // Example default value
        this.available = false; // Example default value
    }
}

File: Backend/src/main/java/com/hotel/demo/models/Room.java
package com.hotel.demo.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Entity
public class Room {
    // Getters and setters
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int roomNumber;
    private String roomType;
    private double price;
    private boolean available;
    // Other attributes and relationships

    // Define getters and setters for other attributes and relationships here
}
